nomad/job_endpoint.go

```diff
@@ -290,10 +290,22 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 		return err
 	}
 
+	// Lookup the job again to get it's version.
+	snap, err = j.srv.State().Snapshot()
+	if err != nil {
+		return err
+	}
+	currentJob, err := snap.JobByID(ws, args.RequestNamespace(), args.Job.ID)
```

Comment:
Would love to keep atomicity here if we can.  If another job update is made in another API call, `currentJob` may be the result of the other update rather than this one.  I would suggesting having the `JobRegisterRequestType` handler return the job version or the updated job.